src/main/java/org/jvnet/hudson/plugins/shelveproject/ShelveProjectAction.java

```diff
@@ -54,7 +54,7 @@ public boolean isShelvingProject() {
   @POST
   public HttpResponse doShelveProject()
           throws IOException, ServletException {
-    Jenkins.getInstance().checkPermission(Item.DELETE);
+    getItem().checkPermission(Item.DELETE);
```

Comment:
If you protect against item being null line 34 (impossible in the production code, but some tests are using it), then you should also protect here